doc/api/esm.md

```diff
@@ -1214,6 +1214,68 @@ loaded from disk but before Node.js executes it; and so on for any `.coffee`,
 `.litcoffee` or `.coffee.md` files referenced via `import` statements of any
 loaded file.
 
+### Register loaders programatically
+
+> The loaders API is being redesigned. This function may disappear or its
+> signature may change. Do not rely on the API described below.
+
+* `specifier` {string} The loader specifier to be registered
+* Returns: {void}
+
+Instead of using the `--experimental-loader http-to-https` option in the CLI, loaders can be registered programatically using the `registerLoader` function.
+
+```js
+import { registerLoader } from 'node:module';
+
+await registerLoader('http-to-https');
+
+// Loader hooks haven't been executed yet.
+
+await import('./my-app.mjs'); // Same behavior of CLI, hooks will be executed before importing the module.
+```
+
+In the example above, we are registering the `http-to-https` loader, but it will only be available in the next imported module, in this case in `my-app.mjs`.
+
+The `--experimental-loader` option of the CLI could still be used together the `registerLoader` function, the loaders registered with the function will follow the same execution chain, but loaders registered within the CLI, will also be executed when importing your loader registered with the function.
+
+```shell
+node \
+  --experimental-loader unpkg \
+  --experimental-loader http-to-https \
+  --experimental-loader cache-buster \
+  entrypoint.mjs
+```
+
+```js
+# entrypoint.mjs
+import { URL } from 'node:url'
+import { registerLoader } from 'node:module'
+
+const relativePathToLoader = new URL('./my-programatically-loader.mjs', import.meta.url).href
+
+await registerLoader(relativePathToLoader)
+await import('./my-app.mjs')
+```
+
+The `my-programatically-loader.mjs` could use all the resources provided by `unpkg`, `http-to-https` and `cache-buster` loaders.
+
+It's also possible to use `registerLoader` more than once.
+
+```js
+# entrypoint.mjs
+import { URL } from 'node:url'
+import { registerLoader } from 'node:module'
+
+const firstLoader = new URL('./first-loader.mjs', import.meta.url).href
+const secondLoader = new URL('./second-loader.mjs', import.meta.url).href
+
+await registerLoader(firstLoader)
+await registerLoader(secondLoader)
+await import('./my-app.mjs')
+```
+
+The `first-loader.mjs` can use all the resources provided by the loaders registered in the CLI. And the `second-loader.mjs` can also use all the resources provided by the loaders registered in the CLI, but not from `first-loader.mjs`, both it will be available only in `my-app.mjs`.
```

Comment:
“But _not”_ from the first loader? Are they not chained? Is this not the equivalent of `--loader first-loader.mjs --loader second-loader.mjs`?

Reply:
Yes, it's the same of `--loader first-loader.mjs --loader second-loader.mjs`. I hadn't noticed, it's reversed. `first-loader.mjs` cannot use the `second-loader.mjs` resources because both are registered in the same module, this means that they will only be available in the next module import (`my-app.mjs`).